zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/GroupQueryController.java

```diff
@@ -55,6 +65,26 @@ public ResponseEntity<GroupSearchQueryResponse> searchGroups(
         .fold(RestErrorMapper::mapProblemToResponse, this::search);
   }
 
+  @PostMapping(
+      path = "/{groupKey}/users/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<UserSearchResponse> searchUsersInGroup(
+      @PathVariable final long groupKey,
+      @RequestBody(required = false) final UserSearchQueryRequest query) {
+    final var groupMemberKeys =
```

Comment:
I think this belongs to a different layer. It should be possible to get users by group programmatically, so it should be handled within a method either in UserServices or GroupServices. But as the strategy is specific to the limitations of ES/OS it actually should be encapsulated by the search client implementation so that the RDBMS implementation can use a plain join. 
There was a bit of a [discussion](https://camunda.slack.com/archives/C06UYJMMETZ/p1733489819132029?thread_ts=1733475623.559369&cid=C06UYJMMETZ) (with no conclusion) on Friday.

Reply:
I agree that this should be moved to the service layer , there is a similar example for searching variables under user task scope [here](https://github.com/camunda/camunda/blob/e8369a7cc2a6b84d7c3534e2150dc9f209d6fbba/service/src/main/java/io/camunda/service/UserTaskServices.java#L153).

I think functionally it makes sense to first fetch the group, I expect we get:
- a 404 when a group with the provided groupKey does not exist. 
- a 403 if the caller does not have READ permission on this group
This would fetch the group member keys for free for ES, maybe it will be more costly for RDBMS.

I have no quick idea how to build a flexible strategy to apply group member keys filter, but we can do it later.

Reply:
>a 404 when a group with the provided groupKey does not exist.
a 403 if the caller does not have READ permission on this group

AFAIK, these are automatically mapped when an exception is thrown from the client response.